preprocess/pre_process.py

```diff
@@ -1,17 +1,60 @@
-import pandas
+import pandas as pd
+import numpy as np
 
-def clean_data(location, method):
-    """
-
-    :param location: String location of raw file
-    :return: data frame of cleaned data
-    """
-
-    df_raw = pandas.read_csv(location)
-
-    """
-    add logic to clean data
-    """
-    df_clean = df_raw #just for testing
-
-    return df_clean
\ No newline at end of file
+def clean_data(location,method="drop"):
+# assigning the NA values for default values
+    missingvalues=["NA","N/A","na","n/a"]
+    df_raw = pd.read_csv(location,na_values=missingvalues,header=None, skiprows=1)
+#   print the raw data after defaulting the na values
+    print ('****RAW dataset****')
+    print (df_raw)
+#  converting all the data to float and cleaning any non numeric values
+    df_raw=df_raw.replace('[^0-9]','',regex=True).astype(float)
+#   print the data with only numeric values 
+    print ('****Removing the non numeric characters****')
+    print (df_raw)
+# method called in if loop for drop or avg
+    
+    if method=="drop":
+        df_method=df_raw.dropna(how='any')
+        print("****Printing the values after filtering the NA****")
+    elif method=="avg":
+        df_method=df_raw.fillna(df_raw.mean())
+        print("****Printing the values after filling the NA values with average of the column************")
+    df_method=df_method.reset_index(drop=True)
+    print (df_method)
+    stds = 1.0
+    lenloc=len(df_method.columns)
+    print ('****Total Number of columns present in dataset:****')
+    print (lenloc)
+    if lenloc==1:
+        df_method.columns=['Y']
+        df_outliers=df_method
+        print('****univariant********')
+    elif lenloc==2:
+        df_method.columns=['Y','X1']
```

Comment:
Dont take hardcoded column names.. they can be anything...  1st one is dependent variable and rest are independent

Reply:
we are renaming what ever the column names to Y,X1,X2,X3, the output will have the same Y,X1,X2,X3 as column names